src/server/server.js

```diff
@@ -107,6 +107,51 @@ const createStock = quote => {
   };
 };
 
+const calculateDateYearsInPast = years => {
+  const year = new Date().getFullYear() - years;
+  const date = new Date();
+  date.setFullYear(year);
+  return date;
+};
+
+const padSingleDigitWithZero = value => {
+  if (typeof value !== 'number' && typeof value !== 'string') {
+    throw new Error(`${padSingleDigitWithZero.name} requires a number or string`);
```

Comment:
`${padSingleDigitWithZero.name} requires a number or numeric string`

Reply:
Also for these formatting functions (formatDate as well) where the argument is expected to be a specific type, you should throw [TypeError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError) when it fails due to wrong type.  More specific errors are always better :)